File: appli-android/app/src/main/java/com/example/sportapp/core/network/AvailableEquipmentApi.kt
package com.example.sportapp.core.network

import com.example.sportapp.core.data.model.AvailableEquipment
import retrofit2.http.*

interface AvailableEquipmentApi {

    @GET("available-equipments")
    suspend fun getAll(): List<AvailableEquipment>

    @PUT("available-equipments/{uuid}")
    suspend fun upsert(@Path("uuid") uuid: String, @Body equipment: AvailableEquipment)

    @PUT("available-equipments/bulk")
    suspend fun upsertAll(@Body equipments: List<AvailableEquipment>)

    @DELETE("available-equipments/{uuid}")
    suspend fun delete(@Path("uuid") uuid: String)
}

File: appli-web/src/app/core/api/available-equipment-api.ts
import { HttpClient } from '@angular/common/http';
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';
import { AvailableEquipment } from '@core/models/available-equipment.model';
import { API_BASE_URL } from './api.config';

@Injectable({ providedIn: 'root' })
export class AvailableEquipmentApi {
  private readonly http = inject(HttpClient);
  private readonly base = `${API_BASE_URL}/available-equipments`;

  getAll(): Observable<AvailableEquipment[]> {
    return this.http.get<AvailableEquipment[]>(this.base);
  }
  upsert(e: AvailableEquipment): Observable<AvailableEquipment> {
    return this.http.put<AvailableEquipment>(`${this.base}/${e.uuid}`, {
      uuid: e.uuid,
      name: e.name,
      updatedAt: e.updatedAt,
    });
  }
  /** Upsert groupé `PUT /available-equipments/bulk` (miroir upsertAll Android) — 1 requête pour N rows. */
  upsertAll(items: AvailableEquipment[]): Observable<AvailableEquipment[]> {
    return this.http.put<AvailableEquipment[]>(
      `${this.base}/bulk`,
      items.map((e) => ({
        uuid: e.uuid,
        name: e.name,
        updatedAt: e.updatedAt,
      })),
    );
  }
  delete(uuid: string): Observable<unknown> {
    return this.http.delete(`${this.base}/${uuid}`);
  }
}
